test/lib/xTest.php

```diff
@@ -155,6 +151,9 @@ public function run(xTestRunner $testRunner)
     private function assertException(\Exception $e = null)
     {
         if (!is_string($this->expectedException)) {
+            if ($e && !(($e instanceof xTestException) && $e->getCode() == xTestException::FAIL)) {
```

Comment:
We don't have any test case this hand made test suite to really check if your change haven't broken anything else. Better to move it to separate PR.